src/app/components/product-page/product-page.component.ts

```diff
@@ -1,13 +1,22 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
 import { IProductInfo } from 'src/app/interfaces/product-info';
-import { productInfo } from 'src/app/mocks/product-info.mock';
+import { ProductService } from 'src/app/services/product.service';
 
 @Component({
 	selector: 'app-product-page',
 	templateUrl: './product-page.component.html',
 	styleUrls: ['./product-page.component.less'],
 	changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ProductPageComponent {
-	public readonly product: IProductInfo = productInfo;
+export class ProductPageComponent implements OnInit {
+	constructor(@Inject(ProductService) private productService: ProductService) {}
```

Comment:
```suggestion
	constructor(private productService: ProductService) {}
```
Если говорим про конечное решение, то лучше таким синтаксисом пользоваться, чтобы сервис кодом не нагружать